File: leetcode/python/001_two_num_sum.py
#!/usr/bin/env python
# coding=utf-8

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sub_dict = {}
        for indx in range(len(nums)):
            sub = target - nums[indx]
            if sub in sub_dict.keys():
                return sub_dict[sub], indx;
            sub_dict[nums[indx]] = indx
        return -1,-1


if __name__ == "__main__":
    Solution solution;
    print(solution.twoSum([2,7,11,15], 9))

File: leetcode/go/001_two_num_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i := 0; i < len(nums); i++ {
        sub := target - nums[i]
        if _, ok := m[sub]; ok {
            return []int{m[sub], i}
        }
    }
    return nil
}

func main(){
    var nums = [4]int{2,7,11,15}
    var tag = twoSum(nums,9)
    fmt.Println(tag[0],tag[i])
}

